internal/git_fetch_test.go

```diff
@@ -0,0 +1,78 @@
+package internal
+
+import (
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	"github.com/databricks/cli/internal/acc"
+	"github.com/databricks/cli/libs/dbr"
+	"github.com/databricks/cli/libs/git"
+	"github.com/databricks/cli/libs/vfs"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const examplesRepoUrl = "https://github.com/databricks/bundle-examples"
+const examplesRepoProvider = "gitHub"
+
+func assertGitInfo(t *testing.T, info git.GitRepositoryInfo, expectedRoot string) {
+	assert.Equal(t, "main", info.CurrentBranch)
+	assert.NotEmpty(t, info.LatestCommit)
+	assert.Equal(t, examplesRepoUrl, info.OriginURL)
+	assert.Equal(t, expectedRoot, info.WorktreeRoot.Native())
+}
+
+func TestAccFetchRepositoryInfoAPI(t *testing.T) {
+	ctx, wt := acc.WorkspaceTest(t)
+	me, err := wt.W.CurrentUser.Me(ctx)
+	require.NoError(t, err)
+
+	targetPath := acc.RandomName("/Workspace/Users/" + me.UserName + "/testing-clone-bundle-examples-")
+	stdout, stderr := RequireSuccessfulRun(t, "repos", "create", examplesRepoUrl, examplesRepoProvider, "--path", targetPath)
+	t.Cleanup(func() {
+		RequireSuccessfulRun(t, "repos", "delete", targetPath)
+	})
+
+	assert.Empty(t, stderr.String())
+	assert.NotEmpty(t, stdout.String())
+	ctx = dbr.MockRuntime(ctx, true)
+
+	for _, inputPath := range []string{
+		targetPath + "/knowledge_base/dashboard_nyc_taxi",
+		targetPath,
+	} {
+		t.Run(inputPath, func(t *testing.T) {
+			info, err := git.FetchRepositoryInfo(ctx, vfs.MustNew(inputPath), wt.W)
+			assert.NoError(t, err)
+			assertGitInfo(t, info, targetPath)
+		})
+	}
+}
+
+func TestAccFetchRepositoryInfoDotGit(t *testing.T) {
```

Comment:
Can we also add test cases when we fail to fetch the data to make sure we handle these gracefully for both cases?

Reply:
I have added a few more test cases in https://github.com/databricks/cli/pull/1945/files#diff-5f006a5f5548016e5de642bb8d3c003b65b3f652a1b9e9dfe46b319c3a57207d

- reading git info when the path is not part of git folder
- reading git info and encountering a permission error (only for .git case).